include/cuco/detail/open_addressing/open_addressing_ref_impl.cuh

```diff
@@ -100,8 +105,9 @@ class open_addressing_ref_impl {
   using value_type          = typename storage_ref_type::value_type;   ///< Storage element type
   using extent_type         = typename storage_ref_type::extent_type;  ///< Extent type
   using size_type           = typename storage_ref_type::size_type;    ///< Probing scheme size type
-  using iterator            = typename storage_ref_type::iterator;     ///< Slot iterator type
-  using const_iterator = typename storage_ref_type::const_iterator;    ///< Const slot iterator type
+  using key_equal           = KeyEqual;  ///< Type of key equality binary callable
```

Comment:
I played around with this a bit and the problem is a bit more complex, since we need to deduce some of the alias types from the ctor using CTAD. It somehow works for now, but we probably need to revisit this in a separate PR. If I apply your suggestion everything goes up in flames -.-

Reply:
I think I see what you mean, I've encountered similar issues back then.